### PR TITLE
Add support for custom S3Destination

### PR DESCRIPTION
**Pull request by VictorBjelkholm**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -22,3 +22,5 @@ _testmain.go
 *.exe
 *.test
 *.prof
+
+examples/examples
```
Comment (gchaincl) [inline .gitignore]: why is this?

File: container.go [hunk 2]
```diff
@@ -21,6 +31,26 @@ func (b ContainerBuilder) Quality(quality uint) ContainerBuilder {
...
+	buf := make([]byte, size)
+	for i := 0; i < size; i++ {
+		buf[i] = alphanum[rand.Intn(len(alphanum))]
+	}
+	return string(buf)
+}
+
+func (b ContainerBuilder) S3Destination(identifier string, destination *S3Destination) ContainerBuilder {
+	if destination.Key == "" {
+		destination.Key = identifier + "-" + randString(10)
```
Comment (gchaincl) [inline container.go]: I'll prefer to keep this kind of logic outside the client.

File: .gitignore (hunk 1 above)
Comment (VictorBjelkholm) [inline .gitignore]: After running the examples, I end up with that garbage

File: container.go (hunk 2 above)
Comment (VictorBjelkholm) [inline container.go]: All right

**Pull request merged**